### ISSUE TITLE
Filter subdirectories from guppy.stream.

### ISSUE THREAD

**Issue by szymonkups** [therealklanni/git-guppy#45]
I have a use case where I want to filter out files from specified subdirectory:
```
gulp.task('pre-commit', function () {
  return guppy.stream('pre-commit')
    .pipe(gulpFilter([ '**/*.js', '!vendor/**/*.js ]))
    .pipe(jshint())
    .pipe(jshint.reporter(stylish))
    .pipe(jshint.reporter('fail'));
});
```
In this situation `gulpFilter` is not working correctly. I did some quick research, and it looks like `guppy.stream()` returns a stream of vinyl files which have their `base` and `relative` fields set to values that are not fully compatible with `gulpFilter`.

```
const gulp = require( 'gulp' );
const guppy = require( 'git-guppy' )( gulp );
const map = require( 'map-stream' );

guppy.stream( 'pre-commit' )
.pipe( map( ( file, cb ) => {
	console.log( file.base, file.relative );

	cb( null, file );
} ) );

/* EXAMPLE OUTPUT
/home/projects/samples/src file1.js
/home/projects/samples/src/lib file2.js
/home/projects/samples/vendor file3.js
*/
```

This example shows that `file.base` is changing, and `file.relative` always points to the file name only. `gulpFilter` uses `file.relative` for multimatching, so no sub-directory checking can be performed.

Could this be changed/fixed, or maybe there is another way to filter files from specified subdirectory?

**Comment by Reinmar** [comment]
Maybe, like in `gulp.src()` and `gulp-watch` we could also have the `baseDir` option. It could be implemented without breaking backward compatibility and it would work in the same way as in those streams:

* when not specified, paths are relative to common parent (that's at least how it works in `gulp.src()`; I haven't checked `guppy()` but I presume it works this way to),
* when specified all paths are relative to that base dir.

**Comment by therealklanni** [comment]
Hi guys. guppy just wraps gulp.src (which is just a wrapper for vinyl-source-stream), so whatever features gulp supports should work exactly the same with guppy. Have you tried passing the option?

**Comment by therealklanni** [comment]
I guess I'm thinking of the prepare-commit-msg hook. Which git hook are you trying to use with guppy? I want to assume pre-commit, since that's the most common.

The problem with what you're asking, is `.stream` is for when you want the indexed (i.e. changed staged for commit), which should be a cached file in a special location in the .git folder in your project, not the actual file in your project. I believe I am mapping the file contents to the working copy path (so this masks the fact that it is actually being streamed from the index), which may be the source of your confusion when you tried to log the file path.

If you want the working copy (the actual file in your project, not the indexed copy), use `.src` instead. This is detailed in the docs as well.

**Comment by therealklanni** [comment]
As far as relative paths and `gulp-filter`, can you try filtering by absolute path?

```js
.pipe(gulpFilter(['/**/*.js']))
```

Unfortunately, the paths come from an internal call to git, which gives me an absolute path. There may be a quick and easy way to resolve this, but I'll have to look into it later. In the mean time, maybe the above example will work for you?

**Comment by szymonkups** [comment]
Thanks for the fast answer! Unfortunately we need to use files staged for commit. Also, we cannot map by absolute path - our solution will be used by other developers so we cannot assume how the path should look. Please take a look at my pull request.

**Comment by therealklanni** [comment]
Published v1.1.0

**Comment by szymonkups** [comment]
That's perfect. Thanks.
